From 7025930f7b73a9db25e85082f977452dfe51f3d7 Mon Sep 17 00:00:00 2001
From: xianchao Zhang <xianchao.zhang@windriver.com>
Date: Tue, 8 Jun 2010 10:00:13 +0800
Subject: [PATCH] e1000e: add missing phy info routine init

The function pointer get_phy_info for IGP03E1000_E_PHY_ID PHY operation
was missing causing the e1000e driver crashes on a null pointer deference.
This patch get the get_phy_info procedure variable initialized.

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
Integrated-by: xianchao Zhang <xianchao.zhang@windriver.com>
---
 drivers/net/e1000e/ich8lan.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/drivers/net/e1000e/ich8lan.c b/drivers/net/e1000e/ich8lan.c
index 8ddb888..c5dc0a0 100644
--- a/drivers/net/e1000e/ich8lan.c
+++ b/drivers/net/e1000e/ich8lan.c
@@ -316,6 +316,7 @@ static s32 e1000_init_phy_params_ich8lan(struct e1000_hw *hw)
 	case IGP03E1000_E_PHY_ID:
 		phy->type = e1000_phy_igp_3;
 		phy->autoneg_mask = AUTONEG_ADVERTISE_SPEED_DEFAULT;
+		phy->ops.get_phy_info = e1000e_get_phy_info_igp;
 		break;
 	case IFE_E_PHY_ID:
 	case IFE_PLUS_E_PHY_ID:
-- 
1.6.0.4

